girder/web_client/src/views/widgets/ItemListWidget.js

```diff
@@ -140,6 +147,110 @@ var ItemListWidget = View.extend({
+            const onLoadImage = (event) => {
+                if (this.observer) {
+                    this.centerSelected();
+                }
+            };
+
+            this.observer = new MutationObserver((mutations) => {
+                // for every mutation
+                _.each(mutations, (mutation) => {
+                    // If no nodes are added we can return
+                    if (!mutation.addedNodes) {
+                        return;
+                    }
+                    // for every added node
+                    _.each(mutation.addedNodes, (node) => {
+                        if (node.className && node.className.indexOf('g-item-list') !== -1) {
+                            // We want to do a onetime scroll to position if the screen is idle
+                            if (window.requestIdleCallback) {
+                                // Processing time is 2 seconds, but should finish much faster
+                                requestIdleCallback(() => {
+                                    this.centerSelected();
+                                }, { timeout: 2000 });
+                            } else {
+                                this.centerSelected();
+                            }
+                        }
+
+                        // For any images added we wait until loaded and rescroll to the selected
+                        if (_.isFunction(node.getElementsByTagName)) {
+                            const imgs = node.getElementsByTagName('img');
+                            for (let i = 0; i < imgs.length; i++) {
+                                // add the event listener to the onload of each image
+                                imgs[i].addEventListener('load', onLoadImage);
+                            }
+                        }
+                    });
+                });
+            });
+
+            // bind mutation observer to a specific element (probably a div somewhere)
+            this.observer.observe(target.parent()[0], { childList: true, subtree: true });
+
+            // Add in scroll event to monitor the scrollPos to prevent uncessary updates and also kill observer when user scrolls
+            widgetcontainer.bind('scroll.observerscroll', (evt) => {
+                if (this.tempScrollPos !== undefined && this.tempScrollPos !== widgetcontainer[0].scrollTop) {
+                    this.tempScrollPos = widgetcontainer[0].scrollTop;
+                    // If the event detail is not 9999 the scroll is should be a user initiated scroll
+                    if (evt.detail !== 9999) {
+                        if (this.observer) {
+                            widgetcontainer.unbind('scroll.observerscroll');
+                            widgetcontainer.unbind('wheel.observerscroll');
+                            this.observer.disconnect();
+                            this.observer = null;
+                            this.tempScrollPos = undefined;
+                        }
```

Comment:
I think it would be clearer if this and the equivalent block below were in their own function rather than repeated.